fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java

```diff
@@ -167,10 +172,21 @@ public void open() throws IOException {
             List<Column> maxComputeColumns = new ArrayList<>(readColumns);
             maxComputeColumns.removeIf(e -> partitionColumns.contains(e.getName()));
             curReader = session.openArrowRecordReader(start, totalRows, maxComputeColumns, arrowAllocator);
+        } catch (TunnelException e) {
+            if (retryCount > 0 && e.getErrorMsg().contains("TableModified")) {
+                retryCount--;
+                // try to refresh table scan and re-open odps
+                refreshTableScan();
+                open();
+            } else {
+                retryCount = 2;
```

Comment:
Why need to reset `retryCount`?
I thought the MaxComputeJniScanner will be new for each query?

Reply:
avoid to call open twice